### PR TITLE
[BugFix] [API] Pull.API.Convert: Only try to get HeadBranch if HeadRepo exist

### LINKED ISSUE #10025: Pull API returning null
- Gitea version (or commit ref): 51f6a7ab1
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL) 
https://gitea.com/api/swagger#/repository/repoGetPullRequest
```
owner = gitea
repo = go-sdk
index = 217
```

## Description

I deleted my fork of `go-sdk` and now the API call is "failing" because it returns `nil`.  
Note "failing" because it _does_ return a response, but it's `null`.

This is the offending func:  
https://github.com/go-gitea/gitea/blob/3c8a5d8783f23eca06eabc5903183f4a2bca2606/modules/convert/pull.go#L18

Shouldn't be too hard to fix, but we need to distinguish what qualifies returning `nil` rather than as much data as possible, because currently returning `null` in the JSON response means PRs will get missed by services using API calls.

### PR DESCRIPTION
**Pull request by 6543**
close #10025

* remove one pr.Issue.LoadRepo (it is ececuted 2 times)
* prevent nil-pointer exeption on user.APIFormat()
* handle non existing HreadRepo

credits to @jolheiser @zeripath who found the witch part of the code i had to look at :)

### CODE REVIEW

**Comment by 6543** [comment]
the diff looks weard on code bloc L101-129 but I only added a if statement before witch  check if `pr.HeadRepoID != 0`

File: modules/convert/pull.go [hunk 1]
```diff
@@ -99,33 +102,35 @@ func ToAPIPullRequest(pr *models.PullRequest) *api.PullRequest {
...
-		}
-	} else {
-		apiHeadBranchInfo := &api.PRBranchInfo{
-			Name:       pr.HeadBranch,
-			Ref:        pr.HeadBranch,
-			RepoID:     pr.HeadRepoID,
-			Repository: pr.HeadRepo.APIFormat(models.AccessModeNone),
-		}
-		headCommit, err = headBranch.GetCommit()
+	if pr.HeadRepoID != 0 {
```
Comment (lunny) [inline modules/convert/pull.go]: pr.HeadRepo != nil

**Comment by 6543** [comment]
@lunny done

File: modules/convert/pull.go [hunk 2]
```diff
@@ -99,33 +101,35 @@ func ToAPIPullRequest(pr *models.PullRequest) *api.PullRequest {
...
-		}
-	} else {
-		apiHeadBranchInfo := &api.PRBranchInfo{
-			Name:       pr.HeadBranch,
-			Ref:        pr.HeadBranch,
-			RepoID:     pr.HeadRepoID,
-			Repository: pr.HeadRepo.APIFormat(models.AccessModeNone),
-		}
-		headCommit, err = headBranch.GetCommit()
+	if pr.HeadRepo != nil {
```
Comment (zeripath) [inline modules/convert/pull.go:102]: In the case that the head repo is nil maybe we need to give some placeholder information (?) - similar to what the UI gives?

Comment (6543) [reply modules/convert/pull.go:102]: you mean implement a `ghost repo` :D

**Comment by lunny** [comment]
@6543 I think you could add a test append on function `TestPullBranchDelete`. I have tested a broken pull request there.

File: modules/convert/pull.go (hunk 2 above)
Comment (zeripath) [reply modules/convert/pull.go:102]: Maybe?

```go
apiHeadBranchInfo := &api.PRBranchInfo{
    Name: pr.HeadBranch,
    Ref: pr.HeadBranch,
    RepoID: -1,
} 
```

Comment (zeripath) [reply modules/convert/pull.go:102]: In the UI we should really be reporting the repo as "(deleted)"

**Review by zeripath — APPROVED** [review]

File: modules/convert/pull.go [hunk 3]
```diff
@@ -35,11 +35,13 @@ func ToAPIPullRequest(pr *models.PullRequest) *api.PullRequest {
 			return nil
 		}
 	}
-	if pr.HeadRepo == nil {
+	if pr.HeadRepoID != 0 && pr.HeadRepo == nil {
 		pr.HeadRepo, err = models.GetRepositoryByID(pr.HeadRepoID)
 		if err != nil {
-			log.Error("GetRepositoryById[%d]: %v", pr.ID, err)
-			return nil
+			if !models.IsErrRepoNotExist(err) {
```
Comment (lafriks) [inline modules/convert/pull.go]: No need for additional if just use && in line above

**Comment by 6543** [comment]
@lunny test added
@lafriks changed

general Info: while I was creating the test i roun in a nil-pointer-exeption so i added a check on user.APIFormat() <- the test now works without this fix but it might prevent som 500

File: modules/convert/pull.go [hunk 4]
```diff
@@ -23,10 +25,12 @@ func ToAPIPullRequest(pr *models.PullRequest) *api.PullRequest {
 		headCommit *git.Commit
 		err        error
 	)
+
 	if err = pr.Issue.LoadRepo(); err != nil {
-		log.Error("loadRepo[%d]: %v", pr.ID, err)
+		log.Error("pr.Issue.loadRepo[%d]: %v", pr.ID, err)
```
Comment (lunny) [inline modules/convert/pull.go]: loadRepo -> LoadRepo

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
Only a nit https://github.com/go-gitea/gitea/pull/10029#discussion_r373319710

**Comment by 6543** [comment]
by the way should we backport this?

**Comment by techknowlogick** [comment]
@6543 please backport :)

**Pull request merged**